src/comparators/PearsonsCorrelation.py

```diff
@@ -25,14 +25,9 @@ def save_groups(self, _group_cols: list[str], **_groups: dict[str, pd.DataFrame]
 
         for idx, col in enumerate(_group_cols):
             for group_name, group in _groups.items():
-                if len(_group_cols) > 1:
-                    self._comparisons[f"{col}__{group_name}"].extend(
-                        [group[idx]] * comparison_count
-                    )
-                else:
-                    self._comparisons[f"{col}__{group_name}"].extend(
-                        [group] * comparison_count
-                    )
+                self._comparisons[f"{col}__{group_name}"].extend(
```

Comment:
Why is the if else no longer necessary?

Reply:
It's because now the only input is a nested iterable, with two levels of nesting, whereas before the group could have been just one level of nesting. I need to adjust the type hint of this function though.